texturepacker/pluginsrc/com/defold/bob/pipeline/tp/AtlasBuilder.java

```diff
@@ -180,24 +180,58 @@ static private TextureSetLayout.Point createPoint(Info.Point point) {
     static private TextureSetLayout.SourceImage createSprite(Info.Sprite srcSprite) {
         TextureSetLayout.SourceImage out = new TextureSetLayout.SourceImage();
 
-        out.name            = srcSprite.getName();
-        out.rotated         = srcSprite.getRotated();
+        boolean rotated = srcSprite.getRotated();
+
+        out.setName(srcSprite.getName());
+        out.setRotated(rotated);
 
         Info.Rect tightRect   = srcSprite.getFrameRect();
-        Info.Size originalSze = srcSprite.getUntrimmedSize();
+        Info.Size originalSize= srcSprite.getUntrimmedSize();
         // The offset from the top left corner of the image, where to find the tight rect
         Info.Point offset     = srcSprite.getCornerOffset();
 
-        out.rect = new TextureSetLayout.Rectangle(tightRect.getX() - offset.getX(), tightRect.getY() - offset.getY(),
-                                                  originalSze.getWidth(), originalSze.getHeight());
 
-        out.indices = new ArrayList<>(srcSprite.getIndicesList());
-        out.vertices = new ArrayList<>();
+        float x = tightRect.getX() - offset.getX();
+        float y = tightRect.getY() - offset.getY();
+        float width;
+        float height;
+
+        // For legacy reasons, the other bob code wants it already rotated
+        if (rotated)
+        {
+            width = originalSize.getHeight();
+            height = originalSize.getWidth();
+        }
+        else
+        {
+            width = originalSize.getWidth();
+            height = originalSize.getHeight();
+        }
```

Comment:
Another fix that highlights that we still have some more stuff to stream line to avoid extra confusion.